### PR TITLE
Feature/lite 29730 migrate to esm and from webpack to vite

### PR DESCRIPTION
**Pull request by paulinaczybir**
- Use ES Modules instead of CommonJs
- Replace webpack build with vite build
- replace testing library from jest to vitest
- upgrade storybook 8 and add support to work with vite

IMPROVEMENTS in the future:
- bringing back svg mock to not load all icons for tests (didn't find way to replace by vitest config how vite deals with icons)
- storybook initial loading time is surprisingly long :(

### CODE REVIEW

File: components/src/core/injector/core/injectorFactory.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { clone, has } from '~core/helpers';
+import { clone, has } from '~core/helpers.js';
```
Comment (nerimartinez) [inline components/src/core/injector/core/injectorFactory.js:1]: why? just being curious

File: components/src/core/injector/index.spec.js [hunk 2]
```diff
@@ -4,19 +4,19 @@ import Core from './core/Core';
 import injectorFactory from './core/injectorFactory';
```
Comment (nerimartinez) [inline components/src/core/injector/index.spec.js:4]: *****

File: components/src/core/injector/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import Core from './core/Core';
-import injectorFactory from './core/injectorFactory';
-import launch from './core/launcher';
+import Core from './core/Core.js';
+import injectorFactory from './core/injectorFactory.js';
```
Comment (nerimartinez) [inline components/src/core/injector/index.js:2]: in the spec is imported without `.js`. why?

**Review by nerimartinez — DISMISSED** [review]

**Comment by arnaugiralt** [comment]
> IMPROVEMENTS in the future:
> 
> bringing back svg mock to not load all icons for tests (didn't find way to replace by vitest config how vite deals with icons)
> storybook initial loading time is surprisingly long :(

Could you add these improvements as new subtasks in the UI Toolkit JIRA US?

**Review by arnaugiralt — CHANGES_REQUESTED** [review]

File: components/src/core/injector/core/injectorFactory.js (hunk 1 above)
Comment (paulinaczybir) [reply components/src/core/injector/core/injectorFactory.js]: This is not needed anymore, so I deleted it to unify imports. It was leftover from the migrating process. Thanks!

File: components/src/core/injector/index.js (hunk 3 above)
Comment (paulinaczybir) [reply components/src/core/injector/index.js]: This is not needed anymore, so I deleted it to unify imports. It was leftover from the migrating process. Thanks!

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -17,7 +17,7 @@ jobs:
         with:
           node-version: 20
       - name: Install dependencies
-        run: npm ci
+        run: npm i
```
[the remark this answers is not in the document]
Comment (paulinaczybir) [reply .github/workflows/publish.yml]: Thanks, I brought back the npm ci, this change is not relevant anymore.

File: package.json [hunk 5]
```diff
@@ -1,81 +1,63 @@
...
+    "test:coverage": "vitest run --coverage.enabled",
+    "test:watch": "vitest watch",
     "storybook": "storybook dev -p 6006",
     "build-storybook": "storybook build"
   },
   "dependencies": {
     "@cloudblueconnect/material-svg": "^1.0.43",
     "mitt": "^3.0.1",
     "tinycolor2": "^1.6.0",
+    "vite": "^5.1.6",
```
[the remark this answers is not in the document]
Comment (paulinaczybir) [reply package.json]: Done, thanks!

**Review by arnaugiralt — APPROVED** [review]

**Pull request merged**